### PR TITLE
loopd: wait for lnd to be unlocked on startup

### LINKED ISSUE #325: starting in locked lnd state
It seems loopd, if started while lnd is still awaiting unlock, will go into an error state instead of waiting graciously. 

This complicates initialization as we now need to start loopd from a higher level after unlock is successful... as opposed to it simply being ready.

### PR DESCRIPTION
**Pull request by carlaKC**
Fixes #325

Depends on lightninglabs/lndclient/pull/36

#### Pull Request Checklist
- [x] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

**Review by guggero — APPROVED** [review]
Nice fix, LGTM!

**Comment by carlaKC** [comment]
@bhandras I can't seem to request you on this, but it's the easiest way to test out the lndclient pr.

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/golang/protobuf v1.3.2
 	github.com/grpc-ecosystem/grpc-gateway v1.14.3
 	github.com/jessevdk/go-flags v1.4.0
-	github.com/lightninglabs/lndclient v0.11.0-3
+	github.com/lightninglabs/lndclient v1.0.1-0.20210111133012-a2e3bbe2d814
```
Comment (guggero) [inline go.mod:13]: I pushed a tag `v0.11.0-5`, please use that.

Comment (carlaKC) [reply go.mod]: done!

File: loopd/run.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func newListenerCfg(config *Config, rpcCfg RPCConfig) *listenerCfg {
 		getLnd: func(network lndclient.Network, cfg *lndConfig) (
 			*lndclient.GrpcLndServices, error) {
 
-			syncCtx, cancel := context.WithCancel(
+			clCtx, cancel := context.WithCancel(
```
Comment (bhandras) [inline loopd/run.go:85]: very nit, why not `callCtx` ? :)

**Review by bhandras — APPROVED** [review]
LGTM :+1:

File: loopd/run.go (hunk 2 above)
Comment (carlaKC) [reply loopd/run.go]: Originally called this client ctx in lndclient, but decided there were too many things called client :) updated to `callerCtx`

**Pull request merged**